src/System.CommandLine/StringExtensions.cs

```diff
@@ -173,13 +174,81 @@ bool CanBeUnbundled(string arg)
 
                 return arg.StartsWith("-") &&
                        !arg.StartsWith("--") &&
-                       arg.RemovePrefix()
-                          .All(CharacterIsValidOptionAlias);
+                       TryUnbundle(arg.RemovePrefix(), out replacement);
+
+                Token TokenForOptionAlias(char c) =>
+                    argumentDelimiters.Contains(c) 
+                    ? null
+                    : knownTokens.SingleOrDefault(t => 
+                        t.Type == TokenType.Option && 
+                        t.Value.RemovePrefix() == c.ToString());
+                
+                void AddRestValue(List<string> list, string rest)
+                {
+                    if (argumentDelimiters.Contains(rest[0]))
+                    {
+                        list[list.Count - 1] += rest;
+                    }
+                    else
+                    {
+                        list.Add(rest);
+                    }
+                }
+                
+                bool TryUnbundle(string arg, out IEnumerable<string> replacement)
+                {
+                    var lastTokenHasArgument = false;
+                    var builder = new List<string>();
+                    for (var i = 0; i < arg.Length; i++)
+                    {
+                        var token = TokenForOptionAlias(arg[i]);
+                        if (token == null)
+                        {
+                            if (lastTokenHasArgument)
+                            {
+                                // The previous token had an optional argument while the current
+                                // character does not match any known tokens. Interpret this as
+                                // the current character is the first char in the argument.
+                                AddRestValue(builder, arg.Substring(i));
+                                break;
+                            }
+                            else
+                            {
+                                // The previous token did not expect an argument, and the current
+                                // character does not match an option, so unbundeling cannot be
+                                // done.
+                                replacement = null;
+                                return false;
+                            }
+                        }
 
-                bool CharacterIsValidOptionAlias(char c) =>
-                    knownTokens.Where(t => t.Type == TokenType.Option)
-                               .Select(t => t.Value.RemovePrefix())
-                               .Contains(c.ToString());
+                        var option = currentCommand?.Children.GetByAlias(token.Value) as IOption;
+                        builder.Add("-" + arg[i]);
```

Comment:
Because the prefixes are configurable, would it be better to use `configuration.Prefixes.First()` instead of `"-"`?

Reply:
How about `token.Value`, given that I have a known token already?

Reply:
Supporting unbundling only on `"-"` simplifies this as well I think.

Reply:
Would `token.Value` still make more sense?